wpengine/faustjs
# feat: Point getting-started to a live demo URL for a faster new user experience (BH-895)

- Points the getting-started example to https://headlessfw.wpengine.com by default, by setting that URL as a fallback in the framework if no `.env.local` (or env vars) exist.
- Makes creation of `.env.local` and setting `WORDPRESS_URL` and `WP_HEADLESS_SECRET` optional and the recommended step 2, instead of a compulsory step 1. 
- Outputs info in the console if env vars are not set.
- Updates readmes to prompt users to create their app first, then later connect it to their own WP instance. (See new [readme quickstart](https://github.com/wpengine/headless-framework/tree/live-demo#quick-start) and [getting started docs](https://github.com/wpengine/headless-framework/tree/live-demo/docs/getting-started).)

## To test
1. Rename the `examples/getting-started/.env.local` if it exists to prevent it from loading (e.g. `_.env.local`).
2. `npm run dev` from the repo root.

You should see notices like this in the console, but the site will still spin up:

```
getting-started: DEMO MODE: The WORDPRESS_URL environment variable is not set. Using demo site URL of https://headlessfw.wpengine.com. Set WORDPRESS_URL to your WordPress site URL and install the WPGraphQL plugin to see your own content.
getting-started: The WP_HEADLESS_SECRET environment variable is not set. Install the WP Engine Headless plugin and set WP_HEADLESS_SECRET to the “Secret Key” from Settings → Headless to enable post previews.
```

3. Visit http://localhost:3000. You should see demo content from https://headlessfw.wpengine.com/.

If you add a `.env.local` with env vars, you'll need to restart the `npm run dev` process (as documented).

## Notes
Will also need to update docs at https://developers.wpengine.com/docs/headless/guides/getting-started/create-app, but this could wait until the updated package/plugin are published.
